packages/backend/src/clients/EmailClient/EmailClient.ts

```diff
@@ -0,0 +1,89 @@
+import { formatTimestamp, PasswordResetLink } from 'common';
+import * as nodemailer from 'nodemailer';
+import Mail from 'nodemailer/lib/mailer';
+import { AuthenticationType } from 'nodemailer/lib/smtp-connection';
+import { LightdashConfig } from '../../config/parseConfig';
+import { SmptError } from '../../errors';
+import Logger from '../../logger';
+
+type Dependencies = {
+    lightdashConfig: Pick<LightdashConfig, 'smtp'>;
+};
+
+export default class EmailClient {
+    lightdashConfig: Pick<LightdashConfig, 'smtp'>;
+
+    transporter: nodemailer.Transporter | undefined;
+
+    constructor({ lightdashConfig }: Dependencies) {
+        this.lightdashConfig = lightdashConfig;
+
+        if (this.lightdashConfig.smtp) {
+            Logger.debug(`Create email transporter`);
+
+            const auth: AuthenticationType = this.lightdashConfig.smtp.auth
+                .accessToken
+                ? {
+                      type: 'OAuth2',
+                      user: this.lightdashConfig.smtp.auth.user,
+                      accessToken: this.lightdashConfig.smtp.auth.accessToken,
+                  }
+                : {
+                      user: this.lightdashConfig.smtp.auth.user,
+                      pass: this.lightdashConfig.smtp.auth.pass,
+                  };
+
+            this.transporter = nodemailer.createTransport(
+                {
+                    host: this.lightdashConfig.smtp.host,
+                    port: this.lightdashConfig.smtp.port,
+                    secure: this.lightdashConfig.smtp.port === 465, // false for any port beside 465, other ports use STARTTLS instead.
+                    auth,
+                    requireTLS: this.lightdashConfig.smtp.secure,
+                    tls: this.lightdashConfig.smtp.allowInvalidCertificate
+                        ? { rejectUnauthorized: false }
+                        : undefined,
+                },
+                {
+                    from: `"${this.lightdashConfig.smtp.sender.name}" <${this.lightdashConfig.smtp.sender.email}>`,
+                },
+            );
+
+            this.transporter.verify((error: any) => {
```

Comment:
Is it a problem that the transporter is created at app start rather than being created, verified + destroyed for each email? I'm not familiar with nodemailer. I guess it doesn't leave a connection open?

Reply:
> You have to create the transporter object only once. If you already have a transporter object you can use it to send mail as much as you like.

Reference: https://nodemailer.com/usage/

Reply:
I'm just not 100% sure how we should handle the failed verification.